### ISSUE TITLE
Make DNS plugin snaps

### ISSUE THREAD

**Issue by bmw** [certbot/certbot#7672]
Using the approach described at https://github.com/certbot/certbot/blob/master/snap/local/README.md, we should make snaps for all of our DNS plugins.

~~We need to resolve https://github.com/certbot/certbot/issues/7667 and https://github.com/certbot/certbot/issues/7945 first. We may also want to resolve https://github.com/certbot/certbot/issues/7854, https://github.com/certbot/certbot/issues/8064, and https://github.com/certbot/certbot/issues/7946 as well.~~

I think this issue should specifically track making `snapcraft.yml` files for each of our DNS plugins. To try and break the work up a bit, I created a separate issue to track building, testing, and deploying these snaps in CI at https://github.com/certbot/certbot/issues/8041. I'm open to other approaches, but I imagine the first PR here defining (some of the) `snapcraft.yml` files which the reviewer can manually build/test.

As part of this, I think we should use core20 as the base for our DNS plugins. Using core18 seems to work for the dnsimple plugin, but I think we'll hit problems with this approach eventually and I think we'll also want to use core20/Python 3.8 everywhere eventually so I figure we might as well do it that way now.

**Comment by bmw** [comment]
This is completely unblocked now as all potential blockers I wrote about in my initial post have been resolved.

**Comment by bmw** [comment]
This was done in https://github.com/certbot/certbot/pull/8096.
